packages/codegen.go/src/clientFactory.ts

```diff
@@ -9,79 +9,82 @@ import { contentPreamble, formatCommentAsBulletItem, formatParameterTypeName, so
 import { ImportManager } from './imports.js';
 
 
-// Creates the content for all <operation>.go files
+// Creates the content for client_factory.go (ARM only)
 export async function generateClientFactory(codeModel: go.CodeModel): Promise<string> {
-  let result = '';
   // generate client factory only for ARM
-  if (codeModel.type === 'azure-arm' && codeModel.clients) {
-    // the list of packages to import
-    const imports = new ImportManager();
-    
-    // there should be at most one client level param: subscriptionID for ARM, any exception is always a wrong swagger definition that we should fix
-    const allClientParams = new Array<go.Parameter>();
-    for (const clients of codeModel.clients) {
-      for (const clientParam of values(clients.parameters)) {
-        if (values(allClientParams).where(param => param.name === clientParam.name).any()) {
-          continue;
-        }
-        allClientParams.push(clientParam);
+  if (codeModel.type !== 'azure-arm' || codeModel.clients.length === 0) {
+    return '';
+  }
+
+  let result = '';
+  // the list of packages to import
+  const imports = new ImportManager();
+  
+  // there should be at most one client level param: subscriptionID for ARM, any exception is always a wrong swagger definition that we should fix
```

Comment:
Should we make this clear by throwing in that case?

Reply:
Thinking about this more, I wonder if this will always be true. It's pretty typical that there's only `subscriptionID` (or no client params), but I wonder if this is an actual invariant we need to enforce. I'm thinking that, for now, I will just remove the comment.